src/BaseSchema.js

```diff
@@ -406,10 +405,17 @@ const BaseSchema = (
   /**
    * It returns all the schema values
    *
+   * @param {Object} [options] - Options
+   * @param {boolean} [options.isRoot = true] - Is a root level schema
    * @returns {object}
    */
-  valueOf: () => {
+  valueOf: ({ isRoot } = { isRoot: true }) => {
     const { properties, definitions, required, $schema, ...rest } = schema
+
+    if (isRoot && required && !required.every((v) => typeof v === 'string')) {
+      throw new FluentSchemaError("'required' has called on root-level schema, check your calls to require()")
```

Comment:
```suggestion
      throw new FluentSchemaError("'required' has called on root-level schema, check your calls to .required()")
```

Reply:
I just copy pasted this line from here: https://github.com/fastify/fluent-json-schema/blob/2bcc53978437003068388a9ff86e8977bd14c0c4/src/BaseSchema.js#L200

Should I also replace it?

Reply:
Thanks. yes